File: library/src/main/java/name/zeno/android/widget/ClipImageLayout.java
package name.zeno.android.widget;


import android.content.Context;
import android.graphics.Bitmap;
import android.util.AttributeSet;
import android.util.TypedValue;
import android.view.ViewGroup;
import android.widget.RelativeLayout;

public class ClipImageLayout extends RelativeLayout
{

  private ClipImageView  clipView;
  private ClipBorderView borderView;

  private int paddingHorizontal = 60;

  public ClipImageLayout(Context context, AttributeSet attrs)
  {
    super(context, attrs);

    clipView = new ClipImageView(context);
    borderView = new ClipBorderView(context);

    ViewGroup.LayoutParams lp = new LayoutParams(-1, -1);
    this.addView(clipView, lp);
    this.addView(borderView, lp);

    // 计算padding的px
    paddingHorizontal = (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, paddingHorizontal, getResources().getDisplayMetrics());
    clipView.setHorizontalPadding(paddingHorizontal);
    borderView.setHorizontalPadding(paddingHorizontal);
  }

  /**
   * 对外公布设置边距的方法,单位为dp
   */
  public void setHorizontalPadding(int mHorizontalPadding)
  {
    this.paddingHorizontal = mHorizontalPadding;
  }

  /**
   * 裁切图片
   */
  public Bitmap clip()
  {
    return clipView.clip();
  }

  public void setBitmap(Bitmap bitmap)
  {
    clipView.setImageBitmap(bitmap);
  }

}

File: library/src/main/java/name/zeno/android/widget/ClipImageLayout.kt
package name.zeno.android.widget


import android.content.Context
import android.graphics.Bitmap
import android.util.AttributeSet
import android.util.TypedValue
import android.widget.RelativeLayout

class ClipImageLayout(context: Context, attrs: AttributeSet) : RelativeLayout(context, attrs) {

  private val clipView: ClipImageView = ClipImageView(context)
  private val borderView: ClipBorderView = ClipBorderView(context)

  private var paddingHorizontal = 60

  init {
    val lp = RelativeLayout.LayoutParams(-1, -1)
    this.addView(clipView, lp)
    this.addView(borderView, lp)

    // 计算padding的px
    paddingHorizontal = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, paddingHorizontal.toFloat(), resources.displayMetrics).toInt()
    clipView.setHorizontalPadding(paddingHorizontal)
    borderView.setHorizontalPadding(paddingHorizontal)
  }

  /**
   * 对外公布设置边距的方法,单位为dp
   */
  fun setHorizontalPadding(mHorizontalPadding: Int) {
    this.paddingHorizontal = mHorizontalPadding
  }

  /**
   * 裁切图片
   */
  fun clip(): Bitmap {
    return clipView.clip()
  }

  fun setBitmap(bitmap: Bitmap) {
    clipView.setImageBitmap(bitmap)
  }

}
